server/src/main/java/org/opensearch/action/bulk/TransportShardBulkAction.java

```diff
@@ -442,7 +443,7 @@ protected long primaryOperationSize(BulkShardRequest request) {
 
     @Override
     public ReplicationMode getReplicationMode(IndexShard indexShard) {
-        if (indexShard.isRemoteTranslogEnabled()) {
+        if (indexShard.indexSettings().isRemoteNode()) {
```

Comment:
The semantics looks weird, like how is an index shard or for that matter an index settings a remote node?

Reply:
We could rename it as `isOnRemoteNode` . Would that help ?

Reply:
IndexShard.isPresentOnRemoteNode() or may be calling `RemoteStoreNodeAttribute.isRemoteDataAttributePresent(IndexShard)` seems like a good option to me.

Reply:
Will take this in a follow up PR .